crypto/include/internal/ec_int.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Internal EC functions for other submodules: not for application use */
+
+#ifndef HEADER_OSSL_EC_INTERNAL_H
+# define HEADER_OSSL_EC_INTERNAL_H
+# include <openssl/opensslconf.h>
+
+# ifndef OPENSSL_NO_EC
+
+#  ifdef __cplusplus
+extern "C" {
+#  endif
+
+#  include <openssl/ec.h>
+
+/*-
+ * Computes the multiplicative inverse of x in the range
+ * [1,EC_GROUP::order).
+ *
+ *         res := x^(-1) (mod EC_GROUP::order)
+ *
+ * - When the EC group order is even, this function returns an error.
+ * - When the EC group order is otherwise composite, the correctness of
+ *   of the output is not guaranteed.
+ * - When x is outside the range [1, EC_GROUP::order), the correctness
+ *   of the output is not guaranteed.
+ */
```

Comment:
@bbbrumley Should I just go ahead and simply say:
```
 * This function expects that the EC_GROUP order is prime and that x is
 * included in the range [1, EC_GROUP::order).
 * In any other case, the correctness of its output is not guaranteed.
```

Reply:
This should be the contract between the EC module and other submodules about this function, given that other submodules should probably not even know (or threat as an opaque object) what an `EC_METHOD` is and if it has a specialized implementation with looser requirements or not.

Reply:
I would say `If the EC_GROUP order is even` so there is no ambiguity of which "group" we're referring to. `group->order` is the cardinality of the subgroup generated by G.